wrong char signedness on non-x86/x86_64

example build on armv7:
```
[183/282] Compiling C++ object src/test-algs.p/test-algs.cc.o
ninja: job failed: g++ -Isrc/test-repacker.p -Isrc -I../src -I. -I.. -I/usr/include/freetype2 -I/usr/include/libpng16 -I/usr/include/glib-2.0 -I/usr/lib/glib-2.0/include -fdiagnostics-color=always -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Wnon-virtual-dtor -std=c++11 -fno-rtti -O0 -fno-exceptions -fno-rtti -fno-threadsafe-statics -fvisibility-inlines-hidden -DHAVE_CONFIG_H -Os -fomit-frame-pointer -Os -fomit-frame-pointer -fPIE -pthread -Wno-non-virtual-dtor -DMAIN -UNDEBUG -MD -MQ src/test-repacker.p/test-repacker.cc.o -MF src/test-repacker.p/test-repacker.cc.o.d -o src/test-repacker.p/test-repacker.cc.o -c ../src/test-repacker.cc
../src/test-repacker.cc: In function 'void start_lookup(int8_t, int8_t, hb_serialize_context_t*)':
../src/test-repacker.cc:115:8: error: narrowing conversion of 'type' from 'int8_t' {aka 'signed char'} to 'char' [-Werror=narrowing]
  115 |     0, type, // type
      |        ^~~~
../src/test-repacker.cc:117:8: error: narrowing conversion of 'num_subtables' from 'int8_t' {aka 'signed char'} to 'char' [-Werror=narrowing]
  117 |     0, num_subtables, // num subtables
      |        ^~~~~~~~~~~~~
cc1plus: some warnings being treated as errors
```

this is because https://github.com/harfbuzz/harfbuzz/blob/f1f2be776bcd994fa9262622e1a7098a066e5cf7/src/test-repacker.cc#L110-L118  

takes a int8_t and casts it to char- which is unsigned outside of x86*

--- a/src/test-repacker.cc
+++ b/src/test-repacker.cc
@@ -112,9 +112,9 @@ static void start_lookup (int8_t type,
                           hb_serialize_context_t* c)
 {
   char lookup[] = {
-    0, type, // type
+    0, (char)type, // type
     0, 0, // flag
-    0, num_subtables, // num subtables
+    0, (char)num_subtables, // num subtables
   };
 
   start_object (lookup, 6, c);

